pkg/collector/osmlogs_collector.go

```diff
@@ -67,21 +67,36 @@ func (collector *OsmLogsCollector) Collect() error {
 	if err != nil {
 		return err
 	}
-
+	resources := []string{"configmap", "serviceaccount", "service", "endpoint", "ingress"}
 	for _, meshName := range meshList {
 		namespacesInMesh, err := getResourceList("namespaces", "openservicemesh.io/monitored-by="+meshName, "-o=jsonpath={..name}", " ")
 		if err != nil {
 			return err
 		}
-		collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName)
+
 		collectControllerLogs(collector, rootPath, meshName)
+		if err != nil {
+			return err
+		}
+
+		collectNamespaceMetadata(collector, namespacesInMesh, rootPath, meshName)
+		if err != nil {
+			return err
+		}
+
+		for _, resource := range resources {
+			err = collectResource(collector, resource, rootPath, meshName)
+			if err != nil {
+				return err
+			}
```

Comment:
This is great!
One thing that you _could_ do if you choose to - you could move the `if err ...` to the previous line so 

```go
err = collectResource(collector, resource, rootPath, meshName)
if err != nil {
```
would become:
```go
if err := collectResource(collector, resource, rootPath, meshName); err != nil {
```